### PR TITLE
Fix for #1136 - case insensitive comparison to detect app service env

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

**Comment by jennyf19** [comment]
thanks @bgavrilMS will take a look later. probably won't merge until after we have the auth schemes done.

File: tests/Microsoft.Identity.Web.Test/WebAppExtensionsTests.cs [hunk 1]
```diff
@@ -634,7 +704,7 @@ private async Task AddMicrosoftIdentityWebApp_TestB2cSpecificSetup(IServiceColle
 
             await remoteFailureFuncMock.ReceivedWithAnyArgs().Invoke(Arg.Any<RemoteFailureContext>()).ConfigureAwait(false);
             // Assert issuer is updated to non-default user flow
-            Assert.Contains(TestConstants.B2CEditProfileUserFlow, redirectContext.ProtocolMessage.IssuerAddress);
+            Assert.Contains(TestConstants.B2CEditProfileUserFlow, redirectContext.ProtocolMessage.IssuerAddress, System.StringComparison.OrdinalIgnoreCase);
```
Comment (jmprieur) [inline tests/Microsoft.Identity.Web.Test/WebAppExtensionsTests.cs:707]: Is it part of this PR?

**Review by jmprieur — COMMENTED** [review]
Thanks @bgavrilMS 
LGTM

One change does not seem to be related to this PR?

File: tests/Microsoft.Identity.Web.Test/WebAppExtensionsTests.cs [hunk 2]
```diff
@@ -411,6 +422,63 @@ public void AddMicrosoftIdentityWebApp_AddsInMemoryTokenCaches()
...
+            try
+            {
+                // Arrange
+                Environment.SetEnvironmentVariable(
+                    AppServicesAuthenticationInformation.AppServicesAuthEnabledEnvironmentVariable, "tRue");
+
+                Environment.SetEnvironmentVariable(
+                    AppServicesAuthenticationInformation.AppServicesAuthIdentityProviderEnvironmentVariable,
+                    "azureactivedirectory"); // used to be uppercase
+
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Web.Test/WebAppExtensionsTests.cs:438]: nit: it used to be camelcase

File: tests/Microsoft.Identity.Web.Test/WebAppExtensionsTests.cs [hunk 3]
```diff
@@ -411,6 +422,63 @@ public void AddMicrosoftIdentityWebApp_AddsInMemoryTokenCaches()
...
+            finally
+            {
+                ResetAppServiceEnv();
+            }
+        }
+
+        [Fact]
+        public void AddMicrosoftIdentityWebApp_AppServices()
+        {
+            try
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Web.Test/WebAppExtensionsTests.cs:457]: combine these two tests into one and use the `[Theory]
 [InlineData(string appServicesEnvEnabled, string idpEnvValue)]`

File: tests/Microsoft.Identity.Web.Test/WebAppExtensionsTests.cs (hunk 1 above)
Comment (jennyf19) [reply tests/Microsoft.Identity.Web.Test/WebAppExtensionsTests.cs:683]: it's just a test

**Review by jennyf19 — APPROVED** [review]

**Pull request merged**